### PR TITLE
Use AndroidX tests runner

### PR DESCRIPTION
**Pull request by Dexterp37**
This PR does the following:

- consolidate all the version for the dependencies in a single file, to make it easy to check if things are out of sync [with a-c](https://github.com/mozilla-mobile/fenix/blob/master/buildSrc/src/main/java/Dependencies.kt);
- replaces legacy testing libraries with AndroidX ones (see [here](https://github.com/dector/android-components/commit/d0b8fb632dd838fc9fe38667d1e58b89c2212a75) for a reference changeset)

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: glean-core/android/src/test/java/mozilla/telemetry/glean/private/EventMetricTypeTest.kt [hunk 1]
```diff
@@ -215,11 +215,8 @@ class EventMetricTypeTest {
         Glean.setUploadEnabled(false)
         assertEquals(false, Glean.getUploadEnabled())
         eventMetric.record(mapOf(testNameKeys.testName to "event2"))
-        try {
```
Comment (badboy) [inline glean-core/android/src/test/java/mozilla/telemetry/glean/private/EventMetricTypeTest.kt]: Removing this and expecting the NPE means everything after that won't run, doesn't it?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,7 +1,6 @@
 #Tue Jun 19 12:19:50 AEST 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.3.1-bin.zip
```
Comment (badboy) [inline gradle/wrapper/gradle-wrapper.properties]: oh noes, a gradle upgrade?^^

**Comment by badboy** [comment]
/home/circleci/project/glean-core/android/src/test/java/mozilla/telemetry/glean/private/EventMetricTypeTest.kt:29:1: Unused import

File: glean-core/android/src/test/java/mozilla/telemetry/glean/private/EventMetricTypeTest.kt (hunk 1 above)
Comment (Dexterp37) [reply glean-core/android/src/test/java/mozilla/telemetry/glean/private/EventMetricTypeTest.kt]: Oh, good catch.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (Dexterp37) [reply gradle/wrapper/gradle-wrapper.properties]: Eheheh yes, to match up a-c :)

File: build.gradle [hunk 3]
```diff
@@ -7,9 +7,33 @@
...
+        android_gradle_plugin: '3.4.1',
+        android_maven_publish_plugin: '3.6.2',
+        coroutines: '1.2.2',
+        dokka: '0.9.17',
+        jna: '5.2.0',
+        junit: '4.12',
+        mockito: '2.24.5',
+        mockwebserver: '3.10.0',
+        kotlin: '1.3.40',
+        robolectric: '4.2.1', // This is different than a-c, but we're fine, it's only tests.
```
[the remark this answers is not in the document]
Comment (Dexterp37) [reply build.gradle]: Yes, the Robolectric runner is no longer used. However, we're still using `shadowOf` in a few places. The a-c PR I've basically copied over had the same problem. I will file a bug to remove this.

File: glean-core/android/build.gradle [hunk 4]
```diff
@@ -162,17 +164,17 @@ dependencies {
...
-    testImplementation 'androidx.test:core-ktx:1.1.0'
-    testImplementation 'com.squareup.okhttp3:mockwebserver:3.10.0'
-    testImplementation "org.mozilla.components:lib-fetch-okhttp:$android_components_version"
-    testImplementation "org.mozilla.components:support-test:$android_components_version"
-    testImplementation "androidx.work:work-testing:2.0.1"
-
-    androidTestImplementation 'com.android.support.test:runner:1.0.2'
-    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+    testImplementation "androidx.test.ext:junit:$versions.androidx_junit"
+    testImplementation "org.robolectric:robolectric:$versions.robolectric"
```
[the remark this answers is not in the document]
Comment (Dexterp37) [reply glean-core/android/build.gradle]: We still use `shadowOf` I'm afraid.

File: glean-core/android/src/test/java/mozilla/telemetry/glean/debug/GleanDebugActivityTest.kt [hunk 5]
```diff
@@ -16,6 +15,7 @@ import org.junit.Before
 import org.robolectric.Robolectric
```
[the remark this answers is not in the document]
Comment (Dexterp37) [reply glean-core/android/src/test/java/mozilla/telemetry/glean/debug/GleanDebugActivityTest.kt]: No `shadowOf` that I know of :)

File: glean-core/android/src/test/java/mozilla/telemetry/glean/debug/GleanDebugActivityTest.kt [hunk 6]
```diff
@@ -27,10 +27,10 @@ import org.junit.Rule
 import org.robolectric.Shadows.shadowOf
```
[the remark this answers is not in the document]
Comment (Dexterp37) [reply glean-core/android/src/test/java/mozilla/telemetry/glean/debug/GleanDebugActivityTest.kt]: This seems to be ok in a-c

File: build.gradle (hunk 3 above)
[the remark this answers is not in the document]
Comment (travis79) [reply build.gradle]: Great, it just seems bloated to include two testing frameworks just to use a single feature of one of them.  If we file a bug to look at removing the dependency on Robolectric's `shadowOf` in the future, then that satisfies me.

[the remark this answers is not in the document]
Comment (Dexterp37) [reply build.gradle]: Thanks! I filed [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1572462).

File: gradlew [hunk 7]
```diff
@@ -28,7 +28,7 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+DEFAULT_JVM_OPTS='"-Xmx64m"'
```
Comment (badboy) [inline gradlew]: Doesn't that set it to 64 megabytes? On CI we [set it to 2048m](https://github.com/mozilla/glean/blob/5066c88a60560607413690e7329721d25fdd656c/.circleci/config.yml#L166).

At the very least this should have a comment what it does and why so low

File: gradlew.bat [hunk 8]
```diff
@@ -14,7 +14,7 @@ set APP_BASE_NAME=%~n0
 set APP_HOME=%DIRNAME%
 
 @rem Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-set DEFAULT_JVM_OPTS=
+set DEFAULT_JVM_OPTS="-Xmx64m"
```
Comment (badboy) [inline gradlew.bat]: As above

File: gradlew (hunk 7 above)
Comment (Dexterp37) [reply gradlew]: That's automatically generated by the Gradle thing, a-c has the same. I'm not sure we should change this,

File: gradlew.bat (hunk 8 above)
Comment (Dexterp37) [reply gradlew.bat]: That's automatically generated by the Gradle thing, a-c has the same. I'm not sure we should change this,

File: gradlew (hunk 7 above)
Comment (badboy) [reply gradlew]: Ah well, then, let's leave it untouched.

**Pull request merged**